backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/rules/RulesData.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hyperledger.bpa.impl.rules;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import lombok.*;
+import lombok.extern.slf4j.Slf4j;
+import org.hyperledger.aries.api.connection.ConnectionRecord;
+import org.hyperledger.aries.api.connection.ConnectionState;
+import org.hyperledger.aries.api.connection.ConnectionTheirRole;
+import org.hyperledger.bpa.model.ActiveRules;
+
+import java.util.UUID;
+
+@Slf4j
+@Data
+@Builder
+public class RulesData {
+
+    private UUID ruleId;
+    private Trigger trigger;
+    private Action action;
+
+    @JsonTypeInfo(
+            use = JsonTypeInfo.Id.NAME,
+            property = "type")
+    @JsonSubTypes({
+            @JsonSubTypes.Type(value = Trigger.ConnectionTrigger.class, name = Trigger.CONNECTION_TRIGGER_NAME),
+            @JsonSubTypes.Type(value = Trigger.ProofReceivedTrigger.class, name = Trigger.PROOF_RECEIVED_TRIGGER_NAME)
+    })
+    @NoArgsConstructor
+    public abstract static class Trigger {
+
+        public static final String CONNECTION_TRIGGER_NAME = "connection";
+        public static final String PROOF_RECEIVED_TRIGGER_NAME = "proof_received";
+
+        abstract boolean apply(EventContext ctx);
+
+        @SuppressWarnings("unused")
+        private String type;
+
+        @JsonTypeName(Trigger.CONNECTION_TRIGGER_NAME)
+        @Builder
+        @Data
+        @EqualsAndHashCode(callSuper = true)
+        @NoArgsConstructor
+        @AllArgsConstructor
+        public static class ConnectionTrigger extends Trigger {
+            private ConnectionTheirRole role;
+            private String tag;
+            private String goalCode;
+
+            @Override
+            public boolean apply(EventContext ctx) {
+                ConnectionRecord connRec = ctx.getConnRec();
+                boolean apply = connRec != null && ConnectionState.REQUEST.equals(connRec.getState());
+                if (role != null) {
+                    apply = apply && role.equals(connRec.getTheirRole());
+                }
+                return apply;
+            }
+        }
+
+        @JsonTypeName(Trigger.PROOF_RECEIVED_TRIGGER_NAME)
+        @Builder
+        @Data
+        @EqualsAndHashCode(callSuper = true)
+        @NoArgsConstructor
+        @AllArgsConstructor
+        public static class ProofReceivedTrigger extends Trigger {
+            private String tag;
+            private UUID proofTemplateId;
+
+            @Override
+            public boolean apply(EventContext ctx) {
+                return false;
```

Comment:
maybe add a TODO, that this can be implemented as soon as proof templates are available?